polyfills/Element/prototype/dataset/detect.js

```diff
@@ -0,0 +1,6 @@
+(document.documentElement.dataset &&
+	(
+		Object.getOwnPropertyDescriptor(Element.prototype, 'dataset')  &&
+		Object.getOwnPropertyDescriptor(Element.prototype, 'dataset').get
```

Comment:
Agree. The only solution I can think of is to use `getOwnPropertyDescriptor` and fallback to `__getter__` if it doesn't exist. Do you think that will be ok?